TigranKirakosov/index.js

```diff
@@ -0,0 +1,253 @@
+      });
+      index = ids.indexOf(id);
+      if (index !== -1) {
+        tasks.splice(index, 1);
+      }
+      localStorage.setItem('tasks', JSON.stringify(tasks));
+    },
+    toggleTask: function(id) {
+      var ids, index;
+      ids = tasks.map(function(current) {
+        return current.id;
+      });
+      index = ids.indexOf(id);
+      if (index !== -1) {
+        Task.prototype.toggleState.call(tasks[index]);
+      }
+      localStorage.setItem('tasks', JSON.stringify(tasks));
+    },
+    clearLS: function() {
+      tasks = [];
+      localStorage.setItem('tasks', []);
+    }
+  };
+})(UIController);
+
+var controller = (function(UICtrl, storageCtrl) {
+  var DOM = UICtrl.getDOMStrings();
+  var setUpEventListeners = function() {
+    document.addEventListener('DOMContentLoaded', ctrlDisplayTasks);
+    document
+      .querySelector(DOM.submitBtn)
+      .addEventListener('click', ctrlAddTask);
+    document.addEventListener('keypress', function(event) {
+      if (event.keyCode === 13 || event.which === 13) {
+        ctrlAddTask();
+        event.preventDefault();
+      }
+    });
+    document
+      .querySelector(DOM.taskList)
+      .addEventListener('click', ctrlDeleteTask);
+    document
+      .querySelector(DOM.taskList)
+      .addEventListener('click', ctrlToggleTask);
+    document
+      .querySelector(DOM.filterBtn)
+      .addEventListener('click', ctrlFilterTasks);
+    document
+      .querySelector(DOM.clearBtn)
+      .addEventListener('click', ctrlClearTasks);
+  };
+  var ctrlDisplayTasks = function() {
+    var tasks, taskList;
+    var html = '';
+    tasks = storageCtrl.getFromLS();
+    taskList = document.querySelector(DOM.taskList);
+    tasks.forEach(function(task) {
+      html += `
+      <li class="collection__item" id="item-${task.id}">
+        <span class="text">${task.text}</span>
+        <div class="icons">
+          <i class="${task.iconClass}"></i>
+          <i class="fas fa-backspace"></i>
+        </div>
+        <span class="deadline">Deadline: ${task.deadline}</span>
+      </li>
+      `;
+    });
+    taskList.innerHTML = html;
+  };
+  var ctrlAddTask = function() {
+    var input, newTask;
+    event.preventDefault();
+    input = UICtrl.getInput();
+    if (input.text !== '') {
+      newTask = storageCtrl.addTask(input.text, input.deadline, input.state);
+      storageCtrl.addToLS(newTask);
+      ctrlDisplayTasks();
+      UICtrl.clearFields();
+    }
+  };
+  var ctrlDeleteTask = function(event) {
+    var itemID, splitID, ID;
+    if (event.target.className === DOM.deleteBtn) {
+      itemID = event.target.parentNode.parentNode.id;
+    }
+    if (itemID) {
+      splitID = itemID.split('-');
+      ID = +splitID[1];
+      storageCtrl.deleteFromLS(ID);
+      UICtrl.deleteListItem(itemID);
+    }
+  };
+  var ctrlToggleTask = function(event) {
+    var itemID, splitID, ID, icon;
+    if (event.target.className === DOM.uncheckedBtn || DOM.checkedBtn) {

```

Comment:
if you wanna check that `className` of target equals to `DOM.uncheckedBtn` or `DOM.checkedBtn` you should write another condition, because current condition checks if `className` equals to `DOM.uncheckedBtn` or `DOM.checkedBtn` isn't equal to boolean `false` ( isn't empty string, `undefined`, `0` and so on)

Reply:
In current case, all I want to do is to check if `event.target.className` matches any of these certain classNames. Isn't that correct if there's no match the boolean result is `false`?

Reply:
yeah, it's correct, but I meant that your condition isn't correct - try in console `if (1 === 2 || 3) console.log('oh!')`

Reply:
Now I see. To be correct, I should've done code like this, I suppose: `if (1 === 2 || 1 === 3) console.log('oh')`